tasks/libs/common/omnibus.py

```diff
@@ -1,9 +1,167 @@
+            'KERNEL_MATRIX_TESTING_',
+            'KUBERNETES_',
+            'MACOS_GITHUB_',
+            'OMNIBUS_',
+            'POD_',
+            'RELEASE_VERSION',
+            'RPM_',
+            'RUN_',
+            'S3_',
+            'SMP_',
+            'SSH_',
+            'TEST_INFRA_',
+            'USE_',
+            'VAULT_',
+            'WINDOWS_',
+        ]
+        excluded_suffixes = [
+            '_SHA256',
+            '_VERSION',
+        ]
+        excluded_values = [
+            "AVAILABILITY_ZONE",
+            "BENCHMARKS_CI_IMAGE",
+            "BUCKET_BRANCH",
+            "BUNDLER_VERSION",
+            "CHANGELOG_COMMIT_SHA_SSM_NAME",
+            "CLANG_LLVM_VER",
+            "CHANNEL",
+            "CI",
+            "COMPUTERNAME" "CONSUL_HTTP_ADDR",
+            "DOGSTATSD_BINARIES_DIR",
+            "EXPERIMENTS_EVALUATION_ADDRESS",
+            "GCE_METADATA_HOST",
+            "GENERAL_ARTIFACTS_CACHE_BUCKET_URL",
+            "GET_SOURCES_ATTEMPTS",
+            "GO_TEST_SKIP_FLAKE",
+            "HOME",
+            "HOSTNAME",
+            "HOST_IP",
+            "INSTALL_SCRIPT_API_KEY_SSM_NAME",
+            "INTEGRATION_WHEELS_CACHE_BUCKET",
+            "IRBRC",
+            "KITCHEN_INFRASTRUCTURE_FLAKES_RETRY",
+            "LESSCLOSE",
+            "LESSOPEN",
+            "LC_CTYPE",
+            "LS_COLORS",
+            "MACOS_S3_BUCKET",
+            "MESSAGE",
+            "OLDPWD",
+            "PROCESS_S3_BUCKET",
+            "PWD",
+            "PYTHON_RUNTIMES",
+            "RESTORE_CACHE_ATTEMPTS",
+            "RUNNER_TEMP_PROJECT_DIR",
+            "RUSTC_SHA256",
+            "RUST_VERSION",
+            "SHLVL",
+            "STATIC_BINARIES_DIR",
+            "STATSD_URL",
+            "SYSTEM_PROBE_BINARIES_DIR",
+            "TRACE_AGENT_URL",
+            "USE_CACHING_PROXY_PYTHON",
+            "USE_CACHING_PROXY_RUBY",
+            "USE_S3_CACHING",
+            "USERDOMAIN",
+            "USERNAME",
+            "USERPROFILE",
+            "VCPKG_BLOB_SAS_URL_SSM_NAME",
+            "WIN_S3_BUCKET",
+            "WINGET_PAT_SSM_NAME",
+            "_",
+            "build_before",
+        ]
+        for p in excluded_prefixes:
+            if key.startswith(p):
+                return False
+        for s in excluded_suffixes:
+            if key.endswith(s):
+                return False
+        if key in excluded_values:
+            return False
+        return True
+
+    return dict(filter(env_filter, sorted(os.environ.items())))
+
+
+def omnibus_compute_cache_key(ctx):
+    print('Computing cache key')
+    h = hashlib.sha1()
+    omnibus_last_commit = ctx.run('git log -n 1 --pretty=format:%H omnibus/', hide='stdout').stdout
+    h.update(str.encode(omnibus_last_commit))
+    print(f'\tLast omnibus commit is {omnibus_last_commit}')
+    buildimages_hash = _get_build_images(ctx)
+    for img_hash in buildimages_hash:
+        h.update(str.encode(img_hash))
```

Comment:
Not needed for now, but for debugging purposes, you may want to log explicitly what buildimages entries go in the cache key, like you do for the environment variables.